src/leaflet.draw.straightlines.js

```diff
@@ -53,6 +53,10 @@
                 var currentLine = getLayerOfType(L.Polyline);
                 if (currentLine) {
                     var latLngs = currentLine.getLatLngs();
+
+                    if (latLngs != undefined && latLngs.length === 1 && latLngs[0].constructor === Array) //then it's a nested array    
```

Comment:
does Array.isArray not work?
it is supported from IE9 onward

this should work:
```
if (latLngs && latLngs.length === 1 && Array.isArray(latLngs[0]))
```

Reply:
it would, but .constructor seemed like the most common way to test if the type is array... is it not good enough?

Reply:
I could use jQuery's $.isArray... is that better?

Reply:
Array.isArray is part of the ecmascript 5 spec, is safe to depend on that

jQuery is wrong because leaflet is not depending on jQuery, now I need to add jQuery as a depencency to make this work.

Reply:
ok, so should I just use Array.isArray?

Reply:
yep

Reply:
ok